<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ParseCodebaseDialog</class>
 <widget class="QDialog" name="ParseCodebaseDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Parse Codebase</string>
  </property>
  <property name="toolTip">
   <string>Try to reuse an existing code database and update the information based on a newer version of the source code</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Project build directory</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <property name="spacing">
          <number>0</number>
         </property>
         <item>
          <widget class="QLineEdit" name="compileCommandsFolder">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="inputMask">
            <string/>
           </property>
           <property name="placeholderText">
            <string>Where the project's CMakeCache.txt is</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="searchCompileCommands">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset theme="folder-open"/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="projectBuildFolderError">
         <property name="styleSheet">
          <string notr="true">color : red</string>
         </property>
         <property name="text">
          <string>TextLabel</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label">
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="text">
          <string>Project source folder</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="sourceFolderLayout">
         <property name="spacing">
          <number>0</number>
         </property>
         <item>
          <widget class="QLineEdit" name="sourceFolder"/>
         </item>
         <item>
          <widget class="QPushButton" name="sourceFolderSearch">
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset theme="folder-open"/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="projectSourceFolderError">
         <property name="styleSheet">
          <string notr="true">color : red</string>
         </property>
         <property name="text">
          <string>TextLabel</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="ignoreLabel">
         <property name="text">
          <string>Ignore files matching glob pattern, separate patterns by a comma.</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLineEdit" name="ignorePattern">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="placeholderText">
            <string>glob</string>
           </property>
           <property name="clearButtonEnabled">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Non-lakosian sub-directories, separate directories with a comma.</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="nonLakosiansLayout">
         <property name="spacing">
          <number>0</number>
         </property>
         <item>
          <widget class="QLineEdit" name="nonLakosians"/>
         </item>
         <item>
          <widget class="QPushButton" name="nonLakosiansSearch">
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset theme="folder-open"/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="label_x1">
         <property name="text">
          <string>Third party packages mapping (&quot;path&quot;: &quot;pkg&quot; pairs separated with a comma).</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="thirdPartyPkgMappingLayout">
         <property name="spacing">
          <number>0</number>
         </property>
         <item>
          <widget class="QLineEdit" name="thirdPartyPkgMapping">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="thirdPartyPkgMappingBtn">
           <property name="text">
            <string>Edit</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Thread Count (defaults to Ideal Thread Count)</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="threadCount"/>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QRadioButton" name="updateDb">
         <property name="toolTip">
          <string>Uses the current database and scans only the differences on the source and header files</string>
         </property>
         <property name="text">
          <string>Incremental update</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="refreshDb">
         <property name="toolTip">
          <string>Erases the entire database, and reescan everything from scratch</string>
         </property>
         <property name="text">
          <string>Replace</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="runCmake">
         <property name="toolTip">
          <string>Re-run cmake on the build directory to update compile_commands.json. This may delete automatically generated files.</string>
         </property>
         <property name="text">
          <string>Re-run CMake</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="showDbErrors">
         <property name="toolTip">
          <string>Save database generation errors on log files</string>
         </property>
         <property name="text">
          <string>Display database errors on the log files</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="physicalOnly">
         <property name="toolTip">
          <string>A quick scan that extracts the minimum necessary to use the app. More parsing will be done as needed.</string>
         </property>
         <property name="text">
          <string>Parse only physical relations</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="disableLakosianRules">
         <property name="toolTip">
          <string>Disable Lakosian rules while parsing (Check this if you don't follow John Lakos' book rules)</string>
         </property>
         <property name="text">
          <string>Disable Lakosian rules</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="errorText">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Error Text</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QProgressBar" name="progressBar">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="value">
          <number>24</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="progressBarText">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Percentage Text</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QStackedWidget" name="stackedWidget">
       <property name="currentIndex">
        <number>1</number>
       </property>
       <widget class="QWidget" name="page">
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QTextBrowser" name="textBrowser"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="page_2">
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QTabWidget" name="tabWidget">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="currentIndex">
            <number>-1</number>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <property name="spacing">
            <number>0</number>
           </property>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="btnSaveOutput">
             <property name="text">
              <string>Save Output</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QPushButton" name="btnParse">
       <property name="text">
        <string>Parse</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnCancelParse">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btnClose">
       <property name="text">
        <string>Hide</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
